src/pages/MainPage.vue

```diff
@@ -426,15 +425,14 @@ export default {
       }
     },
     setActiveRoad: function () {
-      const roadHash = window.location.hash;
-      if (roadHash.length && roadHash.substring(0, 7) === '#/#road') {
-        const roadRequested = roadHash.substring(7);
+      if (Object.prototype.hasOwnProperty.call(this.$route.query, 'road')) {
+        const roadRequested = this.$route.query.road;
         if (roadRequested in this.roads) {
-          this.$store.commit('setActiveRoad', roadHash.substring(7));
+          this.$store.commit('setActiveRoad', roadRequested);
```

Comment:
I'm not sure the pros and cons of different ways to check for keys in an object, but was `'road' in this.$route.query` insufficient? Or simplifying even further to just one if statement:
```js
        if (this.$route.query.road in this.roads)
```
since it would be undefined if it's not there?